app/modules/ranksystem/ranksystem.py

```diff
@@ -95,5 +89,68 @@ async def on_message(self, message):
             member.save()
 
 
+    @bridge.bridge_command()
+    @commands.guild_only()
+    @commands.has_permissions(manage_roles=True)
+    @discord.option(name="level", description="The level at which to unlock this rank. Must be a number.")
+    @discord.option(name="role", description="The role to apply. Can be the id, name, or role itself.")
+    @discord.option(name="message", description="An optional custom message to send when a {user} reaches this rank {level}.")
+    async def rank_create(self, ctx: BridgeContext, level: int, role: str, message:str = None):
```

Comment:
Have you tried changing the role type to DiscordRole? Ideally Pycord would do the conversion for you; is Pycord being a PITA about it?

If not possible do use DiscordRole, it would be better to extract the converter into a helper method/file.

Reply:
Yeah, I tried having it be DiscordRole, (and DiscordRole|str and DiscordRole|str|int and Discord...). It just refuses to work otherwise. String works already since we're only going to be calling it as a command, and it already matches pretty well as a string, since you can give an id, name or <@&ID> notation.

I've gone ahead and moved the snippet into its own function. We could potentially do the same for member/etc. later on.